### ISSUE TITLE
"Go to type" tool includes .g.i.cs files, should not

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/roslyn#26369]
When I use the &quot;go to type&quot; tool (&quot;Ctrl+,&quot;, then search for &quot;t &lt;typename&gt;&quot;), it's finding hits in .g.i.cs files. That probably shouldn't happen. See screen shot.

_This issue has been moved from https://developercommunity.visualstudio.com/content/problem/74648/go-to-type-tool-includes-gics-files-should-not.html
VSTS ticketId: 457985_
_These are the original issue comments:_

Stephen Watkins on 10/1/2017, 08:45 PM (204 days ago): <p>This is slightly more of a pain than originally stated -- more than just cluttering search results, these .g.i.cs files frequently stay open in a "preview" tab somewhere in the tab well (even after I have confirmed another file from the results list), and then cause "this file has been changed externally" dialogs to show up whenever I build my code. Every time I search for something using Edit.GoToAll, which is extremely often, I have to go make sure there are no spurious preview tabs left open.</p>

masahiko miyasaka on 10/16/2017, 11:35 PM (189 days ago): Agree. When I use Ctrl+T search, this happens so often.This is incredibly stressfull.OK ,sometime it'may be necessary to see auto-generated file to check somthing,  but it shoudn't be happen by default.
Daniel Nordström on 11/8/2017, 02:57 AM (167 days ago): <p>I have this problem to, what is worse is that they always is on top in the result as well.</p>

Anton Zimin on 11/23/2017, 11:20 PM (151 days ago): <p>The issue persists in VS 15.4.4.</p><p><img src="https://developercommunity.visualstudio.com/storage/temp/17440-2017-11-24-1414.png"></p>

Daniel Møgelgaard Pedersen on 4/20/2018, 07:50 AM (4 days ago): <p>This is really annoying in large solutions where you rely on "ctrl+," Find Tool to find files, as the g.i.cs files are often at the top of a result list and you can't see which are g.i.cs files and which aren't.<br>This has been an issue since launch. I don't understand what there is to 'Investigate' as it is a pretty straight forward problem.<br>The issue persists in VS 15.5.4.</p>

_These are the original issue solutions:_
(no solutions)

**Comment by CyrusNajmabadi** [comment]
Somewhat of a dupe of https://github.com/dotnet/roslyn/issues/14930

Note: an easy mitigation is to attempt to sort these files better.  Note: i do think it's valuable to have these items.  I know i like going to the real .g source to figure out what's going on and what's actually being generated.

**Comment by omanuke** [comment]
How do u think about having an option like a checkbox to include those files or not?

**Comment by CyrusNajmabadi** [comment]
> How do u think about having an option like a checkbox to include those files or not?

Yes, that's def something that could be considered.  "Include generated code" would def make sense as an option.

**Comment by CyrusNajmabadi** [comment]
> but in my team it is rare that we need to consult the generated file (in our case g.i.cs).

It's my personal preference to look there myself.  :D

Note: we already have smarts for how we sort results.  For example, if you have results in both "SomeType.vb" and in "SomeType.Nested.vb" we'll sort the former results above the latter results even though the latter file name is lexicographically before the former file name.   This is so searches for 'SomeType' don't sort the other partial types above the main type.  

We could def do something similar here for .g.i.cs

**Comment by Solethia** [comment]
> Note: we already have smarts for how we sort results.
> ...
> We could def do something similar here for .g.i.cs

I think the above solution is sufficient to remove most of the annoyance. There is a secondary problem with the way content from different content are displayed in the search output as mentioned by Anton Zimin. If this problem is addressed by sorting differently, it would still be 'nice' to be able to differentiate between content in .cs files and .g.i.cs files (or similar generated files) in the search output.

A possible solution would be to style *.g.* files differently or somehow displaying the full file ending when showing classes in the output, but i guess then it would lose the abstraction of just outputting 'class' information for class items in the search output.

Either way sorting generated files last would be an improvement.

**Comment by Sebastian-Franz** [comment]
Any progress here?

You could add a new search option, that includes hidden files - if someone really cares about them.

**Comment by CyrusNajmabadi** [comment]
@Sebastian-Franz this is on the backlog. It's not being looked at ahead of any scheduled work. If you're interested in improving things here, we would take a community pr though.  I outlined above a viable strategy we would likely take. 

Thanks!

**Comment by DaveInCaz** [comment]
FWIW there is a Stack Overflow question asking about this same issue: https://stackoverflow.com/q/70800335/3195477

**Comment by Matiquiel** [comment]
If you just show g.i.cs files BELOW the actual useful files for most people that will solve the issue. Current problem is that the first  item shown is the g.i.cs which is the file you don't wanna navigate to 99.9% of the times.

I think files from obj folders should be ignored altogether by default with the possibility to include them

**Comment by szalapski** [comment]
This is a really inconvenient think that is getting much less attention than it deserves, and it strikes me that the fix is easy and obvious.  Could someone take a look to prioritize it?  This is the only think keeping me using Resharper routinely (for its Ctrl+T find that already does this).

**Comment by CyrusNajmabadi** [comment]
@szalapski see: https://github.com/dotnet/roslyn/issues/26369#issuecomment-1026987787

**Comment by dvygolov** [comment]
Sad that after 5(!) years after this topic started, this thing is still not implemented :-(

**Comment by CyrusNajmabadi** [comment]
> Sad that after 5(!) years after this topic started, this thing is still not implemented :-(

@dvygolov fortunately, Roslyn is open source :-). So if you're interested in improving things here, you can always contribute a pr. Thanks!
